op-monitorism/transaction_monitor/monitor.go

```diff
@@ -0,0 +1,411 @@
+    ],
+    "stateMutability": "view",
+    "type": "function"
+}]`
+
+// DisputeGameABI contains just the functions we need from the game contract
+const DisputeGameABI = `[{
+    "inputs": [],
+    "name": "gameType",
+    "outputs": [{"internalType": "uint32", "name": "", "type": "uint32"}],
+    "stateMutability": "view",
+    "type": "function"
+}, {
+    "inputs": [],
+    "name": "rootClaim",
+    "outputs": [{"internalType": "bytes32", "name": "", "type": "bytes32"}],
+    "stateMutability": "pure",
+    "type": "function"
+}, {
+    "inputs": [],
+    "name": "extraData",
+    "outputs": [{"internalType": "bytes", "name": "", "type": "bytes"}],
+    "stateMutability": "pure",
+    "type": "function"
+}]`
+
+func NewMonitor(ctx context.Context, log log.Logger, m metrics.Factory, cfg CLIConfig) (*Monitor, error) {
+	client, err := ethclient.Dial(cfg.L1NodeUrl)
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial L1 node: %w", err)
+	}
+
+	factoryABI, err := abi.JSON(strings.NewReader(DisputeGameFactoryABI))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse factory ABI: %w", err)
+	}
+
+	gameABI, err := abi.JSON(strings.NewReader(DisputeGameABI))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse game ABI: %w", err)
+	}
+
+	mon := &Monitor{
+		log:           log,
+		client:        client,
+		watchConfigs:  make(map[common.Address]WatchConfig),
+		allowedAddrs:  make(map[common.Address]bool),
+		gameVerifiers: make(map[common.Address]*DisputeGameVerifier),
+		startBlock:    cfg.StartBlock,
+
+		transactions: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "transactions_total",
+				Help:      "Total number of transactions",
+			},
+			[]string{"watch_address", "from", "to", "status"},
+		),
+
+		unauthorizedTx: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "unauthorized_transactions_total",
+				Help:      "Number of transactions from unauthorized addresses",
+			},
+			[]string{"watch_address", "from"},
+		),
+
+		thresholdExceededTx: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "threshold_exceeded_transactions_total",
+				Help:      "Number of transactions exceeding allowed threshold",
+			},
+			[]string{"watch_address", "from", "threshold"},
+		),
+
+		ethSpent: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "eth_spent_total",
+				Help:      "Cumulative ETH spent by address",
+			},
+			[]string{"address"},
+		),
+
+		unexpectedRpcErrors: m.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: MetricsNamespace,
+				Name:      "unexpected_rpc_errors_total",
+				Help:      "Number of unexpected RPC errors",
+			},
+			[]string{"section", "name"},
+		),
+	}
+
+	// Initialize filters and verifiers
+	for _, config := range cfg.WatchConfigs {
```

Comment:
Similar to the check functions I recommend having a parameter verification function for each check type. Then you can simply loop through, grab the right verification function, execute it.